plugins/repository-gcs/build.gradle

```diff
@@ -52,3 +57,48 @@ thirdPartyAudit.excludes = [
         'org.apache.log.Hierarchy',
         'org.apache.log.Logger',
 ]
+
+/** A task to start the GoogleCloudStorageFixture which emulates a Google Cloud Storage service **/
+task googleCloudStorageFixture(type: AntFixture) {
+  dependsOn compileTestJava
+  executable = new File(project.runtimeJavaHome, 'bin/java')
+  args '-cp', "${ -> project.sourceSets.test.runtimeClasspath.asPath }",
+          'org.elasticsearch.repositories.gcs.GoogleCloudStorageFixture',
+          baseDir, 'bucket_test'
+}
+
+/** A service account file that points to the Google Cloud Storage service emulated by the fixture **/
+File serviceAccountFile = new File(project.buildDir, "/generated-resources/service_account_test.json")
+task createServiceAccountFile() {
+  dependsOn googleCloudStorageFixture
+  doLast {
+    KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA")
+    keyPairGenerator.initialize(1024)
+    KeyPair keyPair = keyPairGenerator.generateKeyPair()
+    String encodedKey = Base64.getEncoder().encodeToString(keyPair.private.getEncoded())
+
+    serviceAccountFile.parentFile.mkdirs()
+    serviceAccountFile.setText("{\n" +
+            '  "type": "service_account",\n' +
+            '  "project_id": "integration_test",\n' +
+            '  "private_key_id": "' + UUID.randomUUID().toString() + '",\n' +
+            '  "private_key": "-----BEGIN PRIVATE KEY-----\\n' + encodedKey + '\\n-----END PRIVATE KEY-----\\n",\n' +
+            '  "client_email": "integration_test@appspot.gserviceaccount.com",\n' +
+            '  "client_id": "123456789101112130594",\n' +
+            "  \"auth_uri\": \"http://${ -> googleCloudStorageFixture.addressAndPort }/o/oauth2/auth\",\n" +
```

Comment:
I don't think the closures are necessary here, since setText takes String and thus these will immediately be resolved.